### PR TITLE
Add nsf badge

### PR DESCRIPTION
**Pull request by mattebaughman**
# Description

Adding NSF funding badge to readme

## Type of change

- Documentation update

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Parsl - Parallel Scripting Library
 ==================================
-|licence| |build-status| |docs|
+|licence| |build-status| |docs| |funding1|
```
Comment (danielskatz) [inline README.rst:3]:
```suggestion
|licence| |build-status| |docs| |NSF-1550588|
```

File: README.rst [hunk 2]
```diff
@@ -26,6 +26,9 @@ The latest Parsl version available on PyPi is v1.0.0.
 .. |docs| image:: https://readthedocs.org/projects/parsl/badge/?version=stable
    :target: http://parsl.readthedocs.io/en/stable/?badge=stable
    :alt: Documentation Status
+.. |funding1| image:: https://img.shields.io/badge/NSF-1550588-blue.svg
```
Comment (danielskatz) [inline README.rst:29]:
```suggestion
.. |NSF-1550588| image:: https://img.shields.io/badge/NSF-1550588-blue.svg
```

**Comment by danielskatz** [comment]
Can we also add badges for the other related awards?

1550562, 1550528, 1550475, 1550476

**Comment by yadudoc** [comment]
@danielskatz  Let's do that in separate a PR(s) ?

**Pull request merged**